src/__tests__/type-modifiers.js

```diff
@@ -443,6 +443,181 @@ test('{enter} on a button', () => {
   `)
 })
 
+test('{space} on a button', () => {
+  const {element, getEventSnapshot} = setup('<button />')
+
+  userEvent.type(element, '{space}')
+
+  expect(getEventSnapshot()).toMatchInlineSnapshot(`
+    Events fired on: button
+
+    button - pointerover
+    button - pointerenter
+    button - mouseover: Left (0)
+    button - mouseenter: Left (0)
+    button - pointermove
+    button - mousemove: Left (0)
+    button - pointerdown
+    button - mousedown: Left (0)
+    button - focus
+    button - focusin
+    button - pointerup
+    button - mouseup: Left (0)
+    button - click: Left (0)
```

Comment:
I'm thinking this is probably a special case where simulating mousing over and clicking wouldn't make sense right? Because anyone trying to do: `userEvent.type(button, '{space}')` is probably *not* wanting to fire *two* click events... Hmmm... Should we make this a special case?

Reply:
I was asking myself the same question. For the sake of consistency I decided it should behave the same way it does with other elements.

Maybe we could add a section about this to the documentation? So it is clear that the most popular way to test keyboard events on a button would be:

```js
button.focus()
userEvent.type(button, '{space}', { skipClick: true })
```

Another idea would be to use element.focus() instead of userEvent.click(element) as a default for `type`.

The benefit would be, that it is less "magic" and we could get rid of `skipClick` altogether.
Also the `.focus()` would just fire additional events, if the element doesn't have focus already. So it would just be a minimal safety net.

For tests that require a `click` beforehand, it would still be easy to do that:

```js
userEvent.click(input)
userEvent.type(input, 'foo')
```